src/pybind/mgr/cephadm/module.py

```diff
@@ -2141,6 +2141,17 @@ def _apply_service_spec(self, spec: ServiceSpec) -> str:
             raise OrchestratorError('cannot scale %s service below 1' % (
                 spec.service_type))
 
+        if spec.placement.count is not None:
+            host_count = len(self.inventory.keys())
+            if spec.service_type in ['mon', 'mgr']:
+                if spec.placement.count > max(5, host_count):
+                    raise OrchestratorError('Cannot apply %s %s daemons with only %s hosts.' % (
+                        spec.placement.count, spec.service_type, host_count))
+            else:
+                if spec.placement.count > (10 * host_count):
+                    raise OrchestratorError('Cannot apply %s %s daemons with only %s hosts.' % (
+                        spec.placement.count, spec.service_type, host_count))
```

Comment:
We could add a `max_count_per_host` config option that defaults to 10 so that the user can adjust this.

Also, we can include a elif for ``spec.placement.count_per_host``

Reply:
For the elif, do you want a max on what the count_per_host can be as well? Also limited by the config option?

Reply:
@liewegas is the new extra if statement about count_per_host what you were looking for?